app/src/main/java/com/epam/havina/task1_recycler/ItemListProvider.java

```diff
@@ -0,0 +1,63 @@
+package com.epam.havina.task1_recycler;
+
+import android.support.annotation.NonNull;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Random;
+
+/**
+ * Generates random CardItems and NameItems and returns them composed in List,
+ * which can then be used in ItemListAdapter.
+ *
+ * @author Lizaveta Havina
+ */
+final class ItemListProvider {
+
+    private static final int ITEMS_NUMBER = 30;
+    private static final List<Item> itemList = new ArrayList<>();
+    private static final String[] randomStringParts = {
+            "abc", "def", "ghi", "jkl", "mno", "pqr", "stu", "vwx", "yz"
+    };
+    private static final StringBuilder descBuilder = new StringBuilder();
+
+    static List<Item> getItemList() {
+        if (itemList.isEmpty()) {
```

Comment:
Why it is static

Reply:
In order not to recreate ItemList each time activity is recreated. Using static modifier and checking for it non-emptiness we guarantee that it will be created only once for the whole app's lifecycle.